src/echo-app/src/renderer/src/dev-plugins.ts

```diff
@@ -3,6 +3,19 @@
  * All of these plugins have HMR enabled, regardless of whether you only have the plugins folder open
  */
 
-export const getDevPlugins = () => {
-  return [import('character-plugin'), import('stash-plugin'), import('poe-log-plugin')]
+export const importDevPlugin = (pluginName: string) => {
```

Comment:
I'm not sure we need to do this, importing the plugin doesn't really enabled and disable them. We can import the plugins and just selectively call start on the enabled ones. This list of plugins will always just be these 3 example plugins since the list is really just here for people to add the plugin they are working on locally but that will never be committed.

Reply:
The main reason to do this was that I need a way to get a handle on the module and it's entry so that I call it's methods on the `plugin-settings-page`. I'm sure there is another way to do this, but couldn't figure it out.

Reply:
You can't set a dynamic variable as import.

We may have to write the file to the system with the plugins we want to be in dev inv. The HMR should regonize this and will rename the imports. This would be the way. Its a bit complicated but it should work.

Reply:
Its because the HMR changes the module names with static replacement. 
`import('character-plugin')` would be => `import('entry-2416da1sd')` or something like this.

Reply:
FWIW HMR still worked with the solution of importing then one by one.

The switch statement that maps the stored plug-in configs to imports allows us to pseudo dynamically import.

Reply:
Yes, but its hardcoded. With my idea, we could use your plugin to dynamially enable and disable plguins in HMR without this hardcoded switch. What do you think?

Reply:
So, we would let your plugin to be  static enabled and the others are controlled by your plugin

Reply:
Its basically this:
https://github.com/PoeStack/poestack-sage/blob/main/src/echo-app/bump-version.js
But the plugins writes this dynamically.

Reply:
@C3ntraX I think I see what you mean, but I think I need a little more help with how that would be implemented. I think I need you to create a gist of how this would work, so I can understand the pattern better.

@zach-herridge a few options for this PR:
- Merge as is, and iterate on it
- I can revert the dev plugin enablement code, and merge just the production plugin bit, then follow up with the solution @C3ntraX mentioned
- Keep it open

I think no matter what I'll probably need some assistance in implementing the dynamic imports